source/adapters/level_zero/queue.cpp

```diff
@@ -1861,19 +1877,18 @@ ur_result_t ur_queue_handle_t_::createCommandList(
   ze_command_list_handle_t ZeCommandList;
 
   uint32_t QueueGroupOrdinal;
+  ZeStruct<ze_command_list_desc_t> ZeCommandListDesc;
+  if (usingCounterBasedEvents()) {
```

Comment:
Does "using counter based events" override the check for:
if (Device->useDriverInOrderLists() && isInOrderQueue()) ?

Or should we be checking for all of them?

Reply:
nvm, I found the code ie:
  this->counterBasedEventsEnabled = isInOrderQueue() &&
                                    Device->useDriverInOrderLists() &&
                                    useDriverCounterBasedEvents;
                                    
  please disregard.

Reply:
Missed this during my rebase. But when counterbased is enabled then both inorder list and inorder queue will be enabled. Because of this we can just check for Device->useDriverInOrderLists() && isInOrderQueue()